toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -79,11 +118,165 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 		return "", fmt.Errorf("failed to compress %s with zstd:\n%w", partitionRawFilepath, err)
 	}
 
-	// Remove raw file since output partition format is raw-zst.
-	err = os.Remove(partitionRawFilepath)
+	return partitionRawFilepath + ".zst", nil
+}
+
+// Prepend a skippable frame with the metadata payload to the specified partition file.
+func addSkippableFrame(partitionFilepath string, magicNumber uint32, frameSize uint32, payload [4]uint32) (err error) {
+	// Read existing data from the partition file.
+	existingData, err := os.ReadFile(partitionFilepath)
```

Comment:
These are potentially fairly large files. So, reading them into memory could potentially take a non-trivial amount of time and will consume a non-trivial amount of memory.

The fancy way of fixing this is:

1. Open file in Go.
2. Add the skippable frame.
3. Call `zstd` without the `-o` so that it outputs to stdout. Do this by calling [exec.Command](https://pkg.go.dev/os/exec#Command) directly (don't use the functions in the 'shell' package). and set the open file (from step 1) on the [Stdout](https://pkg.go.dev/os/exec#Cmd) field.

This will result in the file handle being passed to the zstd process, so that zstd writes directly to the file.

Reply:
I am a little confused here:
for step 1 I am opening the file in go with `os.OpenFile()`...
for step 2, how would I "add" the skippable frame to the open file? 
for step 3, I am writing to the open file with zstd?

Reply:
Using the alternate solution and it works! Thanks! closing this comment.